### DISCUSSION TITLE
How to get score.sc file for RosettaDNA module?

### DISCUSSION THREAD

**Discussion by intomybioverse** [RosettaCommons/rosetta#209] [Rosetta]
Hello, 

I am working with RosettaDNA application. I am getting the output scores and binding energy as the header information of the output PDB files. It is very difficult to analyze the results when the number of PDB structures generated is very high. **How can I get the score information as a separate score.sc file?** 
I tried using "-out:file:scorefile score.sc" in the command file, however, the score.sc file was not generated. 

My command file, named resfile.flags, is attached. 
[resfile.flags.docx](https://github.com/user-attachments/files/17389663/resfile.flags.docx)

**Comment by roccomoretti** [comment]
Unfortunately, it looks like the special output hooks which the RosettaDNA application implements turns off the scorefile outputting machinery.

The best I can suggest is to use the `grep` command to extract the lines you want from the PDB files. For example, `grep 'Binding energy:' *.pdb` will extract all the lines with 'Binding energy:' to the terminal. If you're looking for a slightly better format, you can redirect to file and replace the initial colon with a space (leaving the filename on its own): `grep 'Binding energy:' *.pdb | sed 's/:/ /' | sort > binding_energy.sc` 

You can do similar sorts of things with other lines -- just find the pattern you need to grep out. You can then use the `join` command (e.g. `join total_score.sc binding_energy.sc > combined.sc`) to combine the files based on that first (filename) column. (Assuming it's sorted. There's also additional options to `join` & `cut` which can allow you to select out only the columns you need, though that might be easier for you in whatever tabular processing program (Excel, Pandas) you're using.)

**Reply by intomybioverse** [reply]
Thank you for your help. But using these commands, I am getting only the binding energy for one file at a time. How can I get the same for all the PDB output files at once in a single score.sc file?

**Reply by roccomoretti** [reply]
The `grep 'Binding energy:' *.pdb | sed 's/:/ /' | sort > binding_energy.sc` should process all the files with a pdb extension in the directory, and put the outputs into a single file. (Here you're supposed to use the wildcard expansion of `*.pdb` directly -- it shouldn't be interpreted as a "insert your specific filename here" convention -- you should literally type the asterisk on the command line.)

**Comment by intomybioverse** [comment]
Thank you for your help. It is working fine!
